internal/controller/api/astarte_controller.go

```diff
@@ -215,23 +216,10 @@ func (r *AstarteReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		},
 	}
 
-	return ctrl.NewControllerManagedBy(mgr).
-		For(&apiv1alpha2.Astarte{}, builder.WithPredicates(pred)).
-		Owns(&appsv1.Deployment{}).
-		Owns(&appsv1.StatefulSet{}).
-		Owns(&batchv1.Job{}).
-		Watches(
-			&source.Kind{Type: &v1.Secret{}},
-			handler.EnqueueRequestsFromMapFunc(tlsSecretToAstarteReconcileRequestFn(r.Client)),
-		).
-		Complete(r)
-}
-
-func tlsSecretToAstarteReconcileRequestFn(c client.Client) func(obj client.Object) []reconcile.Request {
-	return func(obj client.Object) []reconcile.Request {
+	tlsSecretToAstarteReconcileRequestFunc := func(_ context.Context, obj client.Object) []reconcile.Request {
 		ret := []reconcile.Request{}
 		astarteList := &apiv1alpha2.AstarteList{}
-		_ = c.List(context.Background(), astarteList, client.InNamespace(obj.GetNamespace()))
+		_ = r.List(context.Background(), astarteList, client.InNamespace(obj.GetNamespace()))
```

Comment:
```suggestion
		_ = c.List(context.Background(), astarteList, client.InNamespace(obj.GetNamespace()))
```
where does `r` come from?